inject.js

```diff
@@ -527,11 +540,58 @@
           setMark(v);
         } else if (action === 'jump') {
           jumpToMark(v);
+        } else if (action === 'adaptive') {
+          toggleAdoptiveSpeed(v, controller);
         }
       }
     });
   }
 
+  function toggleAdoptiveSpeed(v, controller) {
+    tc.settings.adaptiveSpeed = !tc.settings.adaptiveSpeed;
+    const shadowController = controller.shadowRoot.querySelector('#controller');
+    var button = shadowController.querySelector(`button[data-action="adaptive"]`)
+    button.setAttribute('class', tc.settings.adaptiveSpeed? '' : 'hideButton');
+
+    if(tc.settings.adaptiveSpeed){
+
+      if (!tc.adaptiveSpeedCtx){
+        tc.adaptiveSpeedCtx = new AudioContext();
+        tc.adaptiveSpeedSrc = tc.adaptiveSpeedCtx.createMediaElementSource(v);
+        tc.adaptiveSpeedAnalyserNode = tc.adaptiveSpeedCtx.createAnalyser();
+        tc.adaptiveSpeedSrc.connect(tc.adaptiveSpeedAnalyserNode);
+        tc.adaptiveSpeedAnalyserNode.connect(tc.adaptiveSpeedCtx.destination);
+        tc.adaptiveSpeedAnalyserNode.fftSize = 512;
+      }
+      const myDataArray = new Float32Array(tc.adaptiveSpeedAnalyserNode.frequencyBinCount);
+    
+      function audioSample(){
+        if(!v.paused){ 
+          tc.adaptiveSpeedAnalyserNode.getFloatFrequencyData(myDataArray);
+          var dbValue = Math.max(...myDataArray) + 100;
+          var quietZone = 45;
+          var eps = 4;
```

Comment:
What does `eps` stand for?

Reply:
I Thought of Machine epsilon. https://en.wikipedia.org/wiki/Machine_epsilon 
But i think it should be named different. I just wanted to have a buffer zone so the speed does not jump around all the time.